symbolic-unwind/src/base.rs

```diff
@@ -0,0 +1,175 @@
+//! Basic definitions necessary for stack unwinding.
+use std::convert::TryInto;
+use std::fmt::Debug;
+use std::ops::{Add, Div, Mul, Rem, Sub};
+
+/// Trait that abstracts over the [endianness](https://en.wikipedia.org/wiki/Endianness)
+/// of data representation.
+///
+/// This trait provides no other functionality than a method for testing whether
+/// an endianness is big or little. In particular it does not provide methods for
+/// reading number types the way that similar traits/types in `byteorder` and `gimli` do.
+///
+/// We use our own trait here so we don't need to depend on crates (`byteorder`/`gimli`)
+/// for a minuscule part of their features.
```

Comment:
Oh, that's the only reason?  I don't feel very strongly about this so would be happy to be ignored (plus i didn't try this so might be harder than it is worth it), but my guess is that most code-bases using this will already have byteorder in their dependency tree.  Adding a dependency on "^1" is unlikely to pull in a new crate.  The other advantage is that it's a common and widely used crate so people are more likely to be familiar with it, or if they are not yet familiar with it it is more transferrable knowledge.

Reply:
@jan-auer What do you think about this?

Reply:
Update: Jan and I have agreed to leave it like this for now and possibly revisit the issue later.